src/common/utils/combine/index.ts

```diff
@@ -0,0 +1,39 @@
+import type { Plugin } from '../../types';
+
+export const combine = (plugins: Plugin[]): Required<Plugin> => {
+  return {
+    onHydrate: () => {
+      plugins.forEach(plugin => {
+        if (plugin.onHydrate !== undefined) {
```

Comment:
I'd write these as:
```
if (plugin.onHydrate != null) {
```
Just to be more gentle with somebody setting the method to `null`

Reply:
Also another way to specify this "combination" is to have an API like:
```
plugins.run('onHydrate')
```
That would allow us to expand the supported methods without copy pasting these lines for all methods.